src/oclfpga/oclfpga_driver.cc

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "oclfpga_device.h"
+#include <vta/driver.h>
+#include <tvm/runtime/registry.h>
+#include <string>
+#include <iostream>
+
+#define MEM_ADDR_IDENTIFIER (0x18000000)
```

Comment:
One more question -- where is this derived from? Is this address common to all Intel FPGA PCI-e cards, or is it specific to the given board that is being used?

Reply:
Thank you for the comments. This is an arbitrary value created to differentiate virtual and "physical" offset. We convert offset to virtual address by adding MEM_ADDR_IDENTIFIER to the offset value.
In OpenCL, offset of a memory starts at 0. However, by convention, a valid virtual addresses should not be zero. An identifier could help to avoid returning zero as a virtual address.

Reply:
Thanks. If you could add a comment or two above that would be great.